base/src/main/kotlin/net/corda/v5/base/types/MemberX500Name.kt

```diff
@@ -126,23 +127,27 @@ class MemberX500Name(
 
         private fun toAttributesMap(principal: X500Principal): Map<String, String> {
             val result = mutableMapOf<String, String>()
-            LdapName(principal.toString()).rdns.forEach { rdn ->
-                require(rdn.size() == 1) {
-                    "The RDN '$rdn' must not be multi-valued."
-                }
-                rdn.toAttributes().all.asSequence().forEach {
-                    require(it.size() == 1) {
-                        "Attribute '${it.id}' have to contain only single value."
-                    }
-                    val value = it.get(0)
-                    require(value is String) {
-                        "Attribute's '${it.id}' value must be a string"
+            try {
+                LdapName(principal.toString()).rdns.forEach { rdn ->
+                    require(rdn.size() == 1) {
+                        "The RDN '$rdn' must not be multi-valued."
                     }
-                    require(!result.containsKey(it.id)) {
-                        "Duplicate attribute ${it.id}"
+                    rdn.toAttributes().all.asSequence().forEach {
+                        require(it.size() == 1) {
+                            "Attribute '${it.id}' have to contain only single value."
+                        }
+                        val value = it.get(0)
+                        require(value is String) {
+                            "Attribute's '${it.id}' value must be a string"
+                        }
+                        require(!result.containsKey(it.id)) {
+                            "Duplicate attribute ${it.id}"
+                        }
+                        result[it.id] = value
                     }
-                    result[it.id] = value
                 }
+            } catch (e: InvalidNameException) {
+                throw IllegalArgumentException(e)
```

Comment:
I think the convention is to include the exception message too so that it's not buried or lost:
```kotlin
throw new IllegalArgumentException(e.message, e)
```

Reply:
Updated